sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.Tests/StatsbeatTests.cs

```diff
@@ -0,0 +1,101 @@
+            Statsbeat.s_statsBeat_ConnectionString = null;
+            Statsbeat.SetStatsbeatConnectionString(parsedConectionString.IngestionEndpoint);
+
+            Assert.Equal(Statsbeat.StatsBeat_ConnectionString_EU, Statsbeat.s_statsBeat_ConnectionString);
+        }
+
+        [Theory]
+        [InlineData("australiacentral")]
+        [InlineData("australiacentral2")]
+        [InlineData("australiaeast")]
+        [InlineData("australiasoutheast")]
+        [InlineData("brazilsouth")]
+        [InlineData("brazilsoutheast")]
+        [InlineData("canadacentral")]
+        [InlineData("canadaeast")]
+        [InlineData("centralindia")]
+        [InlineData("centralus")]
+        [InlineData("chinaeast2")]
+        [InlineData("chinaeast3")]
+        [InlineData("chinanorth3")]
+        [InlineData("eastasia")]
+        [InlineData("eastus")]
+        [InlineData("eastus2")]
+        [InlineData("japaneast")]
+        [InlineData("japanwest")]
+        [InlineData("jioindiacentral")]
+        [InlineData("jioindiawest")]
+        [InlineData("koreacentral")]
+        [InlineData("koreasouth")]
+        [InlineData("northcentralus")]
+        [InlineData("qatarcentral")]
+        [InlineData("southafricanorth")]
+        [InlineData("southcentralus")]
+        [InlineData("southeastasia")]
+        [InlineData("southindia")]
+        [InlineData("uaecentral")]
+        [InlineData("uaenorth")]
+        [InlineData("westus")]
+        [InlineData("westus2")]
+        [InlineData("westus3")]
+        public void StatsbeatConnectionStringIsSetBasedOnCustomersConnectionStringEndpointInNonEU(string nonEUEndpoint)
+        {
+            var customer_ConnectionString = $"InstrumentationKey=1aa11111-bbbb-1ccc-8ddd-eeeeffff3333;IngestionEndpoint=https://{nonEUEndpoint}.in.applicationinsights.azure.com/";
+            var parsedConectionString = ConnectionStringParser.GetValues(customer_ConnectionString);
+
+            // Set ConnectionString to null as it is static and may be impacted by other tests.
+            Statsbeat.s_statsBeat_ConnectionString = null;
+            Statsbeat.SetStatsbeatConnectionString(parsedConectionString.IngestionEndpoint);
+
+            Assert.Equal(Statsbeat.StatsBeat_ConnectionString_NonEU, Statsbeat.s_statsBeat_ConnectionString);
+        }
+
+        [Fact]
+        public void StatsbeatIsNotInitializedForUnknownRegions()
+        {
+            var customer_ConnectionString = "InstrumentationKey=1aa11111-bbbb-1ccc-8ddd-eeeeffff3333;IngestionEndpoint=https://foo.in.applicationinsights.azure.com/";
+            var parsedConectionString = ConnectionStringParser.GetValues(customer_ConnectionString);
+
+            // Set ConnectionString to null as it is static and may be impacted by other tests.
+            Statsbeat.s_statsBeat_ConnectionString = null;
+            Statsbeat.SetStatsbeatConnectionString(parsedConectionString.IngestionEndpoint);
+
+            Assert.Null(Statsbeat.s_statsBeat_ConnectionString);
```

Comment:
can this also assert that isEnabled is false?

Reply:
s_isEnabled is private.